### ISSUE TITLE
dot to ONNX format issue using 1.9.1

### ISSUE THREAD

**Issue by rogachevai** [pytorch/pytorch#66763]
## 🐛 Bug

Hey, according to [this pull](https://github.com/pytorch/pytorch/pull/62596), it seems like the issue with spectral normalization and self-attentions was solved. However, using conda torch 1.9.1, I still can't export the model into onnx format. It doesn't work even in case of dot example below. Should I use another version or something else? Thanks. 

## To Reproduce

```
class MatmulModel(torch.nn.Module):
    def forward(self,other):
        input = torch.randn(5, requires_grad=True).to(device)
        return torch.dot(input, other)

x = torch.randn(5, requires_grad=True)
test_mm = MatmulModel()
torch.onnx.export(test_mm, x.to(device), f = 'test.onnx')
```

## Expected behavior

Model would be converted into onnx format

## Environment

PyTorch version: 1.9.1
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Майкрософт Windows 10 Pro
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.11 (default, Aug  6 2021, 09:57:55) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19042-SP0
Is CUDA available: True
CUDA runtime version: 11.0.194

GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 472.12
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] Could not collect
[conda] blas                      2.112                       mkl    conda-forge

[conda] blas-devel                3.9.0              12_win64_mkl    conda-forge

[conda] cudatoolkit               11.1.1               heb2d755_9    conda-forge

[conda] libblas                   3.9.0              12_win64_mkl    conda-forge

[conda] libcblas                  3.9.0              12_win64_mkl    conda-forge

[conda] liblapack                 3.9.0              12_win64_mkl    conda-forge

[conda] liblapacke                3.9.0              12_win64_mkl    conda-forge

[conda] mkl                       2021.4.0           h0e2418a_729    conda-forge

[conda] mkl-devel                 2021.4.0           h57928b3_730    conda-forge

[conda] mkl-include               2021.4.0           h0e2418a_729    conda-forge

[conda] numpy                     1.21.2           py38h089cfbf_0    conda-forge

[conda] pytorch                   1.9.1           py3.8_cuda11.1_cudnn8_0    pytorch

[conda] torchaudio                0.9.1                      py38    pytorch

[conda] torchvision               0.10.1               py38_cu111    pytorch
